openmetadata-connector-core/openmetadata_connector_helper.go

```diff
@@ -0,0 +1,677 @@
+	} else {
+		defer r.Body.Close()
+	}
+
+	// Find the ID for the 'table' entity
+	var tableID string
+
+	typeList, r, err := c.MetadataApi.ListTypes(ctx).Category("entity").Limit(TypeListLengthLimit).Execute()
+	if err != nil {
+		s.logger.Warn().Msg(ErrorInPrepareOpenMetadataForFybrik)
+		s.logger.Warn().Msg("Most likely OpenMetadata is not up yet")
+		return false
+	}
+	defer r.Body.Close()
+
+	for i := range typeList.Data {
+		if *typeList.Data[i].FullyQualifiedName == "table" {
+			tableID = *typeList.Data[i].Id
+			break
+		}
+	}
+
+	if tableID == "" {
+		s.logger.Error().Msg(ErrorInPrepareOpenMetadataForFybrik)
+		s.logger.Error().Msg("Failed to find the ID for entity 'table'")
+		return false
+	}
+
+	// Find the ID for the 'string' type
+	var stringID string
+	typeList, r, err = c.MetadataApi.ListTypes(ctx).Category("field").Limit(TypeListLengthLimit).Execute()
+	if err != nil {
+		s.logger.Error().Msg(fmt.Sprintf("Error when calling `MetadataApi.ListTypes``: %v\n", err))
+		s.logger.Error().Msg(fmt.Sprintf(FullHTTPResponse, r))
+		return false
+	}
+	defer r.Body.Close()
+
+	for i := range typeList.Data {
+		if *typeList.Data[i].FullyQualifiedName == "string" {
+			stringID = *typeList.Data[i].Id
+			break
+		}
+	}
+
+	if stringID == "" {
+		s.logger.Error().Msg(ErrorInPrepareOpenMetadataForFybrik)
+		s.logger.Error().Msg("Failed to find the ID for entity 'string'")
+		return false
+	}
+
+	const STRING = "string"
+	// Add custom properties for tables
+	r1, _ := c.MetadataApi.AddProperty(ctx, tableID).CustomProperty(*client.NewCustomProperty(
+		"The vault plugin path where the destination data credentials will be stored as kubernetes secrets", Credentials,
+		*client.NewEntityReference(stringID, STRING))).Execute()
+	defer r1.Body.Close()
+	r2, _ := c.MetadataApi.AddProperty(ctx, tableID).CustomProperty(*client.NewCustomProperty(
+		"Connection type, e.g.: s3 or mysql", ConnectionType,
+		*client.NewEntityReference(stringID, STRING))).Execute()
+	defer r2.Body.Close()
+	r3, _ := c.MetadataApi.AddProperty(ctx, tableID).CustomProperty(*client.NewCustomProperty(
+		"Name of the resource", Name,
+		*client.NewEntityReference(stringID, STRING))).Execute()
+	defer r3.Body.Close()
+	r4, _ := c.MetadataApi.AddProperty(ctx, tableID).CustomProperty(*client.NewCustomProperty(
+		"Geography of the resource", Geography,
+		*client.NewEntityReference(stringID, STRING))).Execute()
+	defer r4.Body.Close()
+	r5, _ := c.MetadataApi.AddProperty(ctx, tableID).CustomProperty(*client.NewCustomProperty(
+		"Owner of the resource", Owner,
+		*client.NewEntityReference(stringID, STRING))).Execute()
+	defer r5.Body.Close()
+	r6, _ := c.MetadataApi.AddProperty(ctx, tableID).CustomProperty(*client.NewCustomProperty(
+		"Data format", DataFormat,
+		*client.NewEntityReference(stringID, STRING))).Execute()
+	r6.Body.Close()
+
+	return true
+}
+
+const DefaultSleepIntervalMS = 500
```

Comment:
can you move the constants to the start of the file

Reply:
moved to separate file. sorted alphabetically.